cmd/loadgenerator/main.go

```diff
@@ -0,0 +1,222 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/crypto/passphrase"
+	"github.com/algorand/go-algorand/daemon/algod/api/client"
+	generatedV2 "github.com/algorand/go-algorand/daemon/algod/api/server/v2/generated"
+	"github.com/algorand/go-algorand/daemon/algod/api/spec/common"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+const transactionBlockSize = 800
+
+func loadMnemonic(mnemonic string) crypto.Seed {
+	seedbytes, err := passphrase.MnemonicToKey(mnemonic)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Cannot recover key seed from mnemonic: %v\n", err)
+		os.Exit(1)
+	}
+
+	var seed crypto.Seed
+	copy(seed[:], seedbytes)
+	return seed
+}
+
+func main() {
+	var cfg config
+	var err error
+	if cfg, err = loadConfig(); err != nil {
+		fmt.Fprintf(os.Stderr, "unable to load config : %v\n", err)
+		os.Exit(1)
+	}
+	fmt.Printf("Configuration file loaded successfully.\n")
+
+	seed := loadMnemonic(cfg.AccountMnemonic)
+	privateKey := crypto.GenerateSignatureSecrets(seed)
+	publicKey := basics.Address(privateKey.SignatureVerifier)
+
+	fmt.Printf("Spending account public key : %v\n", publicKey.String())
+
+	err = spendLoop(cfg, privateKey, publicKey)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "spend loop error : %v\n", err)
+		os.Exit(1)
+	}
+
+	os.Exit(0)
+}
+
+func isSpendRound(cfg config, round uint64) bool {
+	return cfg.RoundModulator == 0 || ((round+cfg.RoundOffset)%cfg.RoundModulator == 0)
+}
+
+func nextSpendRound(cfg config, round uint64) uint64 {
+	if cfg.RoundModulator == 0 {
+		return round
+	}
+	return ((round+cfg.RoundOffset)/cfg.RoundModulator)*cfg.RoundModulator + cfg.RoundModulator
+}
+
+func spendLoop(cfg config, privateKey *crypto.SignatureSecrets, publicKey basics.Address) (err error) {
+	restClient := client.MakeRestClient(*cfg.ClientURL, cfg.APIToken)
+
+	for {
+		waitForSpendingRound(restClient, cfg)
+		queueFull := generateTransactions(restClient, cfg, privateKey, publicKey)
+		if queueFull {
+			waitForNonSpendingRound(restClient, cfg)
+		}
+	}
+}
+
+func waitForNonSpendingRound(restClient client.RestClient, cfg config) {
+	var nodeStatus generatedV2.NodeStatusResponse
+	var err error
+	for {
+		nodeStatus, err = restClient.Status()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "unable to check status : %v", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		if !isSpendRound(cfg, nodeStatus.LastRound) {
+			return
+		}
+	waitForNextBlock:
```

Comment:
can we replace this goto block with another for loop, or are using goto's generally okay?